compare258.c

```diff
@@ -115,7 +115,7 @@ static inline int32_t compare256_unaligned_32_static(const unsigned char *src0,
 
         if (diff) {
             uint32_t match_byte = __builtin_ctz(diff) / 8;
-            return (int32_t)(len + match_byte);
+            return (uint32_t)(len + match_byte);
```

Comment:
```len``` and ```match_byte``` are both uint32_t, so the cast is superfluous.

Reply:
Fixed